airbyte-db/db-lib/src/main/java/io/airbyte/db/factory/ConnectionFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.factory;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Properties;
+
+public class ConnectionFactory {
+
+  /**
+   * Construct a new {@link Connection} instance using the provided configuration.
+   *
+   * @param username The username of the database user.
+   * @param password The password of the database user.
+   * @param connectionProperties The extra properties to add to the connection.
+   * @param jdbcConnectionString The JDBC connection string.
+   * @return The configured {@link Connection}
+   */
```

Comment:
nit: can we also add to the javadoc that we need this because we run tests in parallel and the usual data source does not have sufficient connection guarantees to prevent creating too many connections

Reply:
Added a comment on the class itself